torch/csrc/jit/autodiff.cpp

```diff
@@ -559,129 +339,9 @@ class GradientHelper {
               nullptr,
               nullptr};
 
-    } else if (node->matches("aten::mm(Tensor self, Tensor mat2) -> Tensor")) {
-      return {grads.at(0).mm(inputs.at(1).t()),
-              inputs.at(0).t().mm(grads.at(0))};
-
-    } else if (
-        node->matches(
-            "aten::expand(Tensor self, int[] size, *, bool implicit) -> Tensor")) {
-      const auto& input_sizes = inputs.at(0).sizes();
-      if (input_sizes.size() == 0)
-        return {grads.at(0).sum(), nullptr, nullptr};
-      auto grad_sizes = node->get<std::vector<int64_t>>(attr::size).value();
-      auto grad = grads.at(0);
-      while (grad_sizes.size() > input_sizes.size()) {
-        grad = grad.sum(0, false);
-        grad_sizes.erase(grad_sizes.begin());
-      }
-      for (size_t i = 0; i < input_sizes.size(); ++i) {
-        if (input_sizes[i] == 1 && grad_sizes[i] > 1) {
-          grad = grad.sum(i, true);
-        }
-      }
-      return {grad, nullptr, nullptr};
-
-    } else if (node->matches("aten::squeeze(Tensor self) -> Tensor")) {
```

Comment:
Squeeze exists here but does not appear to be replaced with anything new in the new PR. I am worried that our testing infrastructure is not catching regressions where something that was differentiable before is now no longer differentiable. Do we have tests that check we do not regress what symbolic gradients work? If not, this PR needs more rigorous testing to ensure we didn't lose support.

I also want to make sure our tests are actually testing the correctness of these formula.  Can you verify that introducing an incorrect formula in one of the ported ops actually cases a test failure?

Reply:
Same thing exists for expand. Are these pathways actually dead?